app/src/main/java/org/koitharu/kotatsu/main/ui/MainActivity.kt

```diff
@@ -337,19 +344,40 @@ class MainActivity : BaseActivity<ActivityMainBinding>(),
 	private fun onSearchOpened() {
 		binding.drawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED)
 		drawerToggle.isDrawerIndicatorEnabled = false
-		// Avoiding shadows on the sides if the color is transparent, so we make the AppBarLayout white/dark
-		binding.appbar.setBackgroundColor(ContextCompat.getColor(this, R.color.color_on_secondary))
-		binding.toolbarCard.cardElevation = 0f
+		// Avoiding shadows on the sides if the color is transparent, so we make the AppBarLayout white/grey/black
+		if (get<AppSettings>().isAmoledTheme && get<AppSettings>().theme == AppCompatDelegate.MODE_NIGHT_YES) {
```

Comment:
This may works wrong if theme depends on system settings (`AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM`)
Use this function (place it into `BaseActivity`):
```
protected fun isDarkAmoledTheme(): Boolean {
    val uiMode = resources.configuration.uiMode
    val isNight = uiMode and Configuration.UI_MODE_NIGHT_MASK == Configuration.UI_MODE_NIGHT_YES
    return isNight && get<AppSettings>().isAmoledTheme
}
```

Reply:
👌